### PR TITLE
Minor refactoring of taskRef validation.

### PR DESCRIPTION
**Pull request by squee1945**
# Changes

Small refactoring on taskRef validation. Part of larger refactoring, taken in chunks.

/kind cleanup

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [X] [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [X] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [X] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [X] Release notes block below has been filled in or deleted (only if no user facing changes)

### CODE REVIEW

File: pkg/apis/pipeline/v1beta1/taskrun_validation.go [hunk 1]
```diff
@@ -42,37 +42,40 @@ func (tr *TaskRun) Validate(ctx context.Context) *apis.FieldError {
...
+		errs = errs.Also(apis.ErrMissingOneOf("taskRef", "taskSpec"))
+	}
+	if ts.TaskRef != nil && ts.TaskSpec != nil {
+		errs = errs.Also(apis.ErrMultipleOneOf("taskRef", "taskSpec"))
 	}
 
-	// Check that one of TaskRef and TaskSpec is present
-	if (ts.TaskRef == nil || (ts.TaskRef != nil && ts.TaskRef.Name == "")) && ts.TaskSpec == nil {
-		errs = errs.Also(apis.ErrMissingField("taskref.name", "taskspec"))
+	// Validate that a taskRef has a name; this may be refactored elsewhere later.
```
[the remark this answers is not in the document]
Comment (squee1945) [reply pkg/apis/pipeline/v1beta1/taskrun_validation.go]: Fixed. I just edited the comment.

**Comment by sbwsg** [comment]
/lgtm

The commits need to be squashed to meet the [commit guidelines](https://github.com/tektoncd/community/blob/main/standards.md#commits) but otherwise this looks ready to go from my pov.

**Comment by squee1945** [comment]
@sbwsg - I have squashed, can you re-lgtm?

**Review by vdemeester — APPROVED** [review]

**Pull request merged**